Deprecated `pydantic.BaseModel.dict()` -> use `model_dump()`

**Describe the bug**
As the unittest action shows, the `dict()` method is deprecated since v2. It recommend to use [`model_dump()`](https://docs.pydantic.dev/dev-v2/usage/serialization/#modelmodel_dump) instead.

**Screenshots**
The logs from the GitHub action:
```console
2023-07-20T08:25:29.1307184Z ../home/runner/work/artiq-proxy/artiq-proxy/test.py:60: PydanticDeprecatedSince20: The `dict` method is deprecated; use `model_dump` instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.0.3/migration/
2023-07-20T08:25:29.1308070Z   test_info["ExperimentClass"].dict()
2023-07-20T08:25:29.1309528Z /opt/hostedtoolcache/Python/3.10.12/x64/lib/python3.10/site-packages/pydantic/main.py:897: PydanticDeprecatedSince20: The `dict` method is deprecated; use `model_dump` instead. Deprecated in Pydantic V2.0 to be removed in V3.0. See Pydantic V2 Migration Guide at https://errors.pydantic.dev/2.0.3/migration/
2023-07-20T08:25:29.1310663Z   warnings.warn('The `dict` method is deprecated; use `model_dump` instead.', DeprecationWarning)
2023-07-20T08:25:29.1594060Z ...
2023-07-20T08:25:29.1594853Z ----------------------------------------------------------------------
2023-07-20T08:25:29.1595344Z Ran 5 tests in 0.048s
2023-07-20T08:25:29.1595600Z 
2023-07-20T08:25:29.1595840Z OK
2023-07-20T08:25:29.2610501Z xvfb-run: error: Xvfb failed to start
2023-07-20T08:25:29.2652404Z ##[error]Process completed with exit code 1.
```

**Suggestion**
Replace `.dict()` -> `.model_dump()`.

--- a/test.py
+++ b/test.py
@@ -57,7 +57,7 @@ class RoutingTest(unittest.TestCase):
             self.assertEqual(response.status_code, 200)
             self.assertEqual(
                 response.json()["ExperimentClass"],
-                test_info["ExperimentClass"].dict()
+                test_info["ExperimentClass"].model_dump()
             )
 
     @mock.patch.dict("main.configs", {"repository_path": "repo_path/"})

